plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -298,13 +303,13 @@ private Map<String, String> icebergTableProperties(String location, Map<String,
         return ImmutableMap.copyOf(icebergTableProperties);
     }
 
-    private Schema toIcebergSchema(List<Column> columns)
+    private Schema toIcebergSchema(List<Column> columns, HiveStorageFormat storageFormat)
     {
         AtomicInteger nextFieldId = new AtomicInteger(1);
         List<Types.NestedField> icebergColumns = new ArrayList<>();
         for (Column column : columns) {
             int index = icebergColumns.size();
-            org.apache.iceberg.types.Type type = toIcebergType(typeManager.getType(getTypeSignature(column.getType())), nextFieldId);
+            org.apache.iceberg.types.Type type = toIcebergType(typeManager.getType(getTypeSignature(column.getType(), MILLISECONDS)), nextFieldId, storageFormat);
```

Comment:
Is the addition of `MILLISECONDS` useful ? I don't think it matters for our current logic since we always convert to microsecond precision in iceberg.
Ideally the precision should come from the source table's column type.

Reply:
Let's drop the addition of MILLISECONDS as parameter, it's not needed

Reply:
ping

Reply:
`getTypeSignature(HiveType type)` is deprecated. I'm fine with adding the argument, but it should be a separated commit because it's not needed for supporting timestamp type.